components/core/src/clp_s/clp-s.cpp

```
@@ -116,6 +171,215 @@ bool compress(CommandLineArguments const& command_line_arguments) {
+    byte_buf.insert(byte_buf.cend(), view.begin(), view.end());
+    serializer.clear_ir_buf();
+}
+
+template <typename encoded_variable_t>
+auto unpack_and_serialize_msgpack_bytes(
+        std::vector<uint8_t> const& msgpack_bytes,
+        Serializer<encoded_variable_t>& serializer
+) -> bool {
+    try {
+        auto const msgpack_obj_handle{msgpack::unpack(
+                clp::size_checked_pointer_cast<char const>(msgpack_bytes.data()),
```

cppcheck

[error] Exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)